source/vistas/core/graphics/mesh.py

```diff
@@ -79,6 +79,13 @@ def __init__(
             glEnableVertexAttribArray(1)
             glVertexAttribPointer(1, 3, GL_FLOAT, GL_FALSE, sizeof(GLfloat) * 3, None)
 
+        if self.has_color_array:
```

Comment:
Ah, in that case it might be possible to do that. The issue is if the underlying data is a string. Serializing that to a float will need some thought. In Envision everything is a numerical value, but for features in general that's likely not to be the case. The `color_function` helps get past some of this complexity, but ties it to the color buffer. Perhaps we could instead have the color function translate to shader logic instead!